civis/tests/test_io.py

```diff
@@ -419,3 +433,24 @@ def test_split_schema_tablename_raises():
     s = "table.with.too.many.periods"
     with pytest.raises(ValueError):
         civis.io._tables.split_schema_tablename(s)
+
+
+@mock.patch.object(civis.io._tables, '_sql_script', autospec=True,
+                   return_value=[-1000, 1000])
+@mock.patch.object(civis.io._tables, 'APIClient', autospec=True)
+def test_export_to_civis_file(mock_APIClient, mock_sql_script):
+    expected = [{'file_id': 9844453}]
+
+    # Mock client.scripts.get_sql_runs to return successful response with data
+    # (mock_client must not have channels in its spec to avoid hitting PubNub)
+    mock_client = mock.Mock(spec=['scripts'])
+    mock_client.scripts = mock.Mock(spec=['get_sql_runs'])
+    response = MockResponse(state='success', output=expected)
+    mock_client.scripts.get_sql_runs.return_value = response
+    mock_APIClient.return_value = mock_client
+
+    sql = "SELECT 1"
+    fut = civis.io.export_to_civis_file(sql, 'fake-db',
+                                        polling_interval=POLL_INTERVAL)
+    data = fut.result()['output']
+    assert data == expected
```

Comment:
Is it worth also checking that we called `mock_client.scripts.post_sql` with the input `sql`? That risks breaking if we come up with a different way to export SQL queries, but that would be a major version change of the API, so maybe it's okay.

Reply:
I added an assert on `_sql_script` rather than `post_sql` and another test for `_sql_script`, which I thought might be more in the spirit of unit testing.